kubevirt_api/tests/test_unit.py

```diff
@@ -0,0 +1,253 @@
+# (C) Datadog, Inc. 2024-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+from copy import deepcopy
+from unittest.mock import MagicMock  # noqa: F401
+
+import pytest
+
+from datadog_checks.dev.utils import get_metadata_metrics
+from datadog_checks.kubevirt_api import KubeVirtApiCheck
+
+from .conftest import mock_http_responses
+from .mock_response import GET_VMIS_RESPONSE, GET_VMS_RESPONSE
+
+pytestmark = [pytest.mark.unit]
+
+healthz_tags = [
+    "endpoint:https://10.244.0.38:443/healthz",
+    "pod_name:virt-api-98cf864cc-zkgcd",
+    "kube_namespace:kubevirt",
+    "kube_cluster_name:test-cluster",
+]
+
+
+def test_check_collects_all_metrics(dd_run_check, aggregator, instance, mocker):
+    mocker.patch("requests.get", wraps=mock_http_responses)
+
+    check = KubeVirtApiCheck("kubevirt_api", {}, [instance])
+
+    check._setup_kube_client = lambda: None
+    check.kube_client = MagicMock()
+    check.kube_client.get_vms.return_value = GET_VMS_RESPONSE["items"]
+    check.kube_client.get_vmis.return_value = GET_VMIS_RESPONSE["items"]
+
+    dd_run_check(check)
+
+    aggregator.assert_metric(
+        "kubevirt_api.can_connect",
+        value=1,
+        tags=healthz_tags,
+    )
+
+    metrics_tags = [
+        "endpoint:https://10.244.0.38:443/metrics",
+        "kube_namespace:kubevirt",
+        "kube_cluster_name:test-cluster",
+        "pod_name:virt-api-98cf864cc-zkgcd",
+    ]
+
+    aggregator.assert_metric("kubevirt_api.process.cpu_seconds.count", tags=metrics_tags)
```

Comment:
nit- could you define a list of expected metrics then iterate through the list and assert each metric has metrics_tags?